### PR TITLE
Stylo - support image-orientation property

### PR DESCRIPTION
**Pull request by chenpighead**
First, we need to make Servo's image-orientation parser to be agreed with Gecko's.
Numbers without any AngleUnit, including unitless 0 angle, should be invalid
for image-orientation. However, rotate() and skew() for transform properties
accept unitless 0 angle. In order to make all these properties work properly,
I fixed Angle::parse() to match Gecko. For the existing users of Angle::parse(),
I create Angle::parse_with_unitless() and use it as an alternative for them.
Once https://github.com/w3c/csswg-drafts/issues/1162 is resolved, we shall be
able to use an unified version of Angle::parse() then.

The parser of image-orientation is also fixed to report parsing errors on
empty string.

Then, with the newly added binding functions support in Gecko side, we shall
reuse the same methods from Gecko to pass the computed value from Servo to Gecko.

Gecko bug: Bug 1341758

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16334)

### CODE REVIEW

**Comment by Wafflespeanut** [comment]
Unit tests need to be updated.

**Comment by chenpighead** [comment]
Yeah, just noticed that and updated the PR.
I wonder why this only happens on Windows....

**Comment by Wafflespeanut** [comment]
Oh, we've disabled some stuff in Travis since it has previously caused problems (#15076, #14723)

**Comment by chenpighead** [comment]
stylo try looks fine: https://treeherder.mozilla.org/#/jobs?repo=try&revision=f6608cc9bd1b117d58a980535560a3a8b7d5401f

r? @emilio

**Comment by emilio** [comment]
@bors-servo r+

Thanks!

File: components/style/properties/gecko.mako.rs [hunk 1]
```diff
@@ -2776,6 +2776,34 @@ fn static_assert() {
...
+            },
+            T::AngleWithFlipped(ref angle, flipped) => {
+                unsafe {
+                    bindings::Gecko_SetImageOrientation(&mut self.gecko, angle.radians() as f64, flipped);
+                }
+            }
+        }
+    }
+
+    pub fn copy_image_orientation_from(&mut self, other: &Self) {
```
Comment (nox) [inline components/style/properties/gecko.mako.rs]: This would be better as `Clone::clone_from`, IMO.

Comment (emilio) [inline components/style/properties/gecko.mako.rs]: We'd need to change all the implementations in the style system though.

Comment (emilio) [inline components/style/properties/gecko.mako.rs]: And this only copies a property, which isn't what that trait is meant to represent, right?

Comment (nox) [inline components/style/properties/gecko.mako.rs]: Err yes, forgot on which type this method is, with the mako stuff. :)

Comment (chenpighead) [inline components/style/properties/gecko.mako.rs]: No idea what `Clone::clone_from` is...
Is this related to the clone_xxx_from()? I thought we need clone only for animatable properties, no?

**Comment by bholley** [comment]
Needs a bindings.rs update. You can grab them as an artifact from your try push.

**Comment by chenpighead** [comment]
Ok, I grab the bindings update from try. Hopefully we could pass the tests now.
@bholley, thanks for the help. :)

**Comment by nox** [comment]
These 2 commits should be squashed together.

**Comment by nox** [comment]
Actually no, but the commit that updates the bindings should be first in the PR.

**Comment by chenpighead** [comment]
Ok, update on the way.

**Comment by chenpighead** [comment]
Looks like just a timeout thing....:/
@bors-servo retry

**Comment by nox** [comment]
@chenpighead What is "a timeout thing"? We usually don't retry PR without filing an intermittent issue for the failing tests, if we consider them intermittent.
